source/common/event/dispatcher_impl.cc

```diff
@@ -117,6 +121,71 @@ DispatcherImpl::createClientConnection(Network::Address::InstanceConstSharedPtr
                                                          std::move(transport_socket), options);
 }
 
+namespace {
+Network::Address::InstanceConstSharedPtr
+nextClientAddress(const Network::Address::InstanceConstSharedPtr& server_address) {
+  uint64_t id = 0;
+  return std::make_shared<Network::Address::EnvoyInternalInstance>(absl::StrCat(server_address->asStringView(), "_", ++id));
+}
+} // namespace
+
+Network::ClientConnectionPtr
+DispatcherImpl::createInternalConnection(Network::Address::InstanceConstSharedPtr internal_address,
+                                         Network::Address::InstanceConstSharedPtr local_address) {
+  ASSERT(isThreadSafe());
+  if (internal_address == nullptr) {
+    return nullptr;
+  }
+  if (local_address == nullptr) {
+    local_address = nextClientAddress(internal_address);
+  }
+  // Find the internal listener callback. The listener will setup the server connection.
+  auto iter = internal_listeners_.find(internal_address->asString());
+  for (const auto& [name, _] : internal_listeners_) {
+    ENVOY_LOG_MISC(debug, "lambdai: p listener {}", name);
```

Comment:
I'm inclined to do my next pass once you've had a chance to clean these up, and do a sweep to make sure the code is commented.

Reply:
I think given the complexity it'd be better to split out what we can.  It could be that it's just 2-3?  As said I think the buffer code would split out pretty easily so let's give that a try and see what's left.
Also w.r.t. logging I think it's fine to leave many of the logs in (all the error corner case logs for example) and just move them from lambdai/LOG_MISC to real logs.  It's fine to tackle that as the last step when the big PR is ready for review if you prefer.